src/buildtools/buildtools.targets

```diff
@@ -1,9 +1,11 @@
 <Project>
 
   <PropertyGroup>
-    <DotNetExe Condition=" '$(OS)' == 'Unix' ">dotnet</DotNetExe>
-    <DotNetExe Condition=" '$(OS)' != 'Unix' ">dotnet.exe</DotNetExe>
-    <!-- When building from source, `dotnet` is expected to be on the path. -->
+    <!-- during a regular build `dotnet(.exe)` is expected to be on the path -->
+    <DotNetExe Condition="'$(OS)' == 'Unix'">dotnet</DotNetExe>
+    <DotNetExe Condition="'$(OS)' != 'Unix'">dotnet.exe</DotNetExe>
+    <!-- if already running under dotnet, use that instance -->
+    <DotNetExe Condition="'$(DOTNET_HOST_PATH)' != ''">$(DOTNET_HOST_PATH)</DotNetExe>
     <DotNetExePath>$(DotNetExe)</DotNetExePath>
   </PropertyGroup>
```

Comment:
Shouldn't be necessary. Arcade defines `DotNetTool` here: [https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/RepoLayout.props#L35](https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/RepoLayout.props#L35)